namespacesmapper/dynakubes.go

```diff
@@ -0,0 +1,134 @@
+package namespacesmapper
+
+import (
+	"context"
+	"fmt"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func updateMapFromDynaKube(dk *dynatracev1alpha1.DynaKube, selector labels.Selector, cfgmap *corev1.ConfigMap, nsList *corev1.NamespaceList) bool {
+	updated := false
+
+	if cfgmap.Data == nil {
+		cfgmap.Data = make(map[string]string)
+	}
+
+	for _, namespace := range nsList.Items {
+		matches := selector.Matches(labels.Set(namespace.Labels))
+		mapdkname, ok := cfgmap.Data[namespace.Name]
+
+		if matches {
+			if !ok || mapdkname != dk.Name {
+				cfgmap.Data[namespace.Name] = dk.Name
+				updated = true
+			}
+		} else {
+			if ok && mapdkname == dk.Name {
+				delete(cfgmap.Data, namespace.Name)
+				updated = true
+			}
+		}
+	}
+	return updated
+}
+
+func doMapFromDynaKube(ctx context.Context, clt client.Client, opns string, cfgmapname string, dk *dynatracev1alpha1.DynaKube, crSelector selectorFunc, nsList *corev1.NamespaceList) error {
+	cfgmap, err := getOrCreateMap(ctx, clt, opns, cfgmapname)
+	if err != nil {
+		return err
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(crSelector(*dk))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if updateMapFromDynaKube(dk, selector, cfgmap, nsList) {
+		if err := clt.Update(ctx, cfgmap); err != nil {
+			return errors.WithMessagef(err, "failed to update %s", cfgmapname)
+		}
+	}
+	return nil
+}
+
+func MapFromDynaKube(ctx context.Context, clt client.Client, opns string, dk *dynatracev1alpha1.DynaKube) error {
+	nsList := &corev1.NamespaceList{}
+	err := clt.List(ctx, nsList)
+
+	if err != nil {
+		return errors.Cause(err)
+	}
+
+	diErr := doMapFromDynaKube(ctx, clt, opns, dataIngestMapName, dk,
+		func(dk dynatracev1alpha1.DynaKube) *metav1.LabelSelector {
+			return dk.Spec.DataIngestSpec.Selector
+		}, nsList)
+
+	cmErr := doMapFromDynaKube(ctx, clt, opns, codeModulesMapName, dk,
+		func(dk dynatracev1alpha1.DynaKube) *metav1.LabelSelector {
+			return dk.Spec.CodeModules.Selector
+		}, nsList)
+
```

Comment:
I don't really like how this file handles mapping for data ingest **and** codemodules while still being specifically made for those two. would it be possible, for example, to abstract common code and then create more specific code for either data ingest and code modules?